app/services/goals/create.rb

```diff
@@ -57,9 +57,13 @@ def goal_create_message(goal)
     end
 
     def create_goal_images(goal:)
-      params[:images].each do |image|
-        goal_image = goal.goal_images.build(image: image[:image_data].as_json)
-        goal_image.image_derivatives!
+      if params[:images] && params[:images].length > 0 && params[:election_selected].nil?
+        params[:images].each do |image|
+          goal_image = goal.goal_images.build(image: image[:image_data].as_json)
+          goal_image.image_derivatives!
+        end
+      else
+        goal.goal_images.build
```

Comment:
I think the following will simplify the method:
```suggestion
      return goal.goal_images.create! if params[:election_selected].present?
      return unless params[:images] && params[:images].length > 0
      
      params[:images].each do |image|
          goal_image = goal.goal_images.build(image: image[:image_data].as_json)
          goal_image.image_derivatives!
      end
```

Reply:
Also, I don't think build creates the record in the database, but I might be wrong

Reply:
I see! All good